### DISCUSSION TITLE
using `set: []` and `connectOrCreate` in one query

### DISCUSSION THREAD

**Discussion by ptrxyz** [prisma/prisma#20990] [Q&A]
### Question

I have an update query to my table:

```ts
prisma.table.update({
  where: { id: baseID },
  data: {
    materials: {
      set: [],
      connectOrCreate: parsedData.materials.map((element) => ({
    where: { id: element.id },
    create: { id: element.id, title: element.title }
      }))
    }
  }
})
```

I am using `set: []` and `connectOrCreate: [...]` in the same query and it seems to matter what order I supply those options.
Is that intended use? Or should I avoid doing it like this?

**Comment by ptrxyz** [comment]
I mean, the funny thing is: when I use `set: [], connectOrCreate: [a,b,c]`, then it works as expected: all connections are removed and then `a, b, c` are connected.
When I swap them to be `connectOrCreate: [a,b,c], set: []`, then I always end up with an empty list (= no connections).
My problem with this is that it seems that JS does not guarantee that object keys stay in the order of insertion, so I might end up in an undefined state.
I am not super familiar with Prisma, is there maybe another way to have `connectOrCreateAndDeleteAllPreviousConnections`/`setAndCreateIfMissing` without having to run multiple queries/use a transaction?

**Reply by ludralph** [reply]
Hi @ptrxyz 👋 

The order of execution of `set` and `connect` does matter.

This was previously discussed in [here](https://github.com/prisma/prisma/discussions/19491).

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by ludralph** [reply]
Hey @ptrxyz ,

do you have any more questions around this issue? If no, it would be really helpful if you could quickly **mark this Discussion as _answered_** to indicate that your issue has been resolved. 

In case you do have any outstanding questions, feel free to let us know and we'd be happy to help!

**Comment by ptrxyz** [comment]
On Discord someone said the it is possible to use `set` and `connect` together and then the order would matter. However I was curious how the order can be preserved if we simply pass around POJOs. To my understanding ECMA does not guarantee that the order of properties on am object is preserves when you iterate over them.

Isn't that a problem?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
